pkgs/tools/filesystems/mkspiffs/default.nix

```diff
@@ -0,0 +1,33 @@
+{ stdenv, fetchgit, git
+, extraBuildFlags ? ""
+, buildConfigName ? "generic" }:
+
+# We can build mkspiffs with custom CPPFLAGS by passing them in extraBuidFlags.
+# The upstream has several named presets for CPPFLAGS whose names will be passed in buildConfigName.
+
+stdenv.mkDerivation rec {
+  name = "mkspiffs-${version}";
+  version = "0.2.3";
+
+  src = fetchgit {
+    url = "https://github.com/igrr/mkspiffs";
```

Comment:
Use `fetchFromGitHub` instead

Reply:
fetchFromGitHub doesn't support deepClone as far as I know.

Reply:
However, for `fetchFromGitHub` there is `fetchSubmodules = true;` which does what I want.